server/src/main/java/io/crate/replication/logical/plan/CreateSubscriptionPlan.java

```diff
@@ -76,12 +76,13 @@ public void executeOrFail(DependencyCarrier dependencies,
 
         var subscribingUser = connectionInfo.settings().get(ConnectionInfo.USERNAME.getKey());
         if (subscribingUser == null || subscribingUser.isEmpty()) {
-            throw new InvalidArgumentException(
+            throw new CreateSubscriptionException(
                 String.format(Locale.ENGLISH, "Setting '%s' must be provided on CREATE SUBSCRIPTION", ConnectionInfo.USERNAME.getKey())
             );
         }
-        for (var setting : settings.names()) {
-            throw new InvalidArgumentException(
+        if (settings.names().isEmpty() == false) {
+            var setting = settings.names().iterator().next();
+            throw new CreateSubscriptionException(
```

Comment:
Do I get it fight that we throw if not empty? Looks like if `CREATE SUB WITH` has anything specified we throw an exception.

Maybe even can ignore first element and just do empty check and throw smth like "CREATE SUB with" is not supported? But not insisting, maybe showing first name is even better

Reply:
It was throwing error for first element already, but I agree to throw a generic error that `WITH ...` is not supported at all, will adjust.